modulemd/v2/modulemd-defaults-v1.c

```diff
@@ -47,6 +48,95 @@ G_DEFINE_TYPE (ModulemdDefaultsV1,
                MODULEMD_TYPE_DEFAULTS)
 
 
+static gboolean
+modulemd_defaults_v1_equals (ModulemdDefaults *self_1, ModulemdDefaults *self_2, GError **error)
+{
+  ModulemdDefaultsV1 *v1_self_1, *v1_self_2;
+  v1_self_1 = NULL;
+  v1_self_2 = NULL;
+  GHashTableIter iter_1, iter_2;
+  gpointer key_1, key_2, value_1, value_2;
+  GHashTable *profile_defaults_1, *profile_defaults_2;
+  GHashTable *intent_default_profiles_1, *intent_default_profiles_2;
+
+
+  g_return_val_if_fail (MODULEMD_IS_DEFAULTS_V1 (self_1), NULL);
+  v1_self_1 = MODULEMD_DEFAULTS_V1 (self_1);  
+  g_return_val_if_fail (MODULEMD_IS_DEFAULTS_V1 (self_2), NULL);
+  v1_self_2 = MODULEMD_DEFAULTS_V1 (self_2);  
+
+  if (!MODULEMD_DEFAULTS_CLASS (modulemd_defaults_v1_parent_class)
+             ->equals (self_1, error);)
+    {
+      return FALSE;
+    }
+
+  if (!MODULEMD_DEFAULTS_CLASS (modulemd_defaults_v1_parent_class)
+             ->equals (self_2, error);)
+    {
+      return FALSE;
+    }
+
+  if(g_strcmp0 (v1_self_1->intent, v1_self_2->intent))
+    {
+      return FALSE;
+    }
+
+  const gchar *default_stream_1 =
+    modulemd_defaults_v1_get_default_stream(v1_self_1, v1_self_1->intent);
```

Comment:
Don't use `v1_self_1->intent` or expect it to exist. I didn't realize it was still in there, but it's nonfunctional and always NULL. I need to remove it. (It was part of an earlier implementation that I scrapped and must have missed that piece).

You should just compare `v1_self_1->default_stream` and `v2_self->default_stream` directly. Don't bother using the getters; this is an internal function.